lib/Core/ExecutorUtil.cpp

```diff
@@ -126,6 +126,30 @@ namespace klee {
     if (numOperands > 1) op2 = evalConstant(ce->getOperand(1), ki);
     if (numOperands > 2) op3 = evalConstant(ce->getOperand(2), ki);
 
+    /* Checking for possible errors during constant folding */
+    switch (ce->getOpcode()) {
+    case Instruction::SDiv:
+    case Instruction::UDiv:
+    case Instruction::SRem:
+    case Instruction::URem:
+      if (op2->getLimitedValue() == 0) {
+	std::string msg("Division/modulo by zero during constant folding at location ");
+	llvm::raw_string_ostream os(msg);
+	os << (ki ? ki->printFileLine().c_str() : "[unknown]");
+	klee_error("%s", os.str().c_str());
```

Comment:
Ah you're right sorry. It looks like all the constant (including constant expression attached to instrutions) get evaluated in `Executor::bindModuleConstants()` where the calls to `getOperandNum()` in `lib/Module/KModule.cpp` are where the constants get stored in the list of constants.

I guess this will have to do for now. A way this could be improved in the future is to bind constant expressions that can't be evaluated into Expr that represent the value but aren't constant folded. This is tricky right now because the Expr constructors are intertwined with constant folding. These two things need to be separated and put into the ExprBuilder so that we can construct expressions that are not constant folded when we need to.